content/docs/rfcs/73/README.md

```diff
@@ -0,0 +1,161 @@
+Counterparties will be able to register members to a RLN membership set without exposing the user's private keys.
+
+## Background
+The [32/RLN-V1](/spec/32) protocol, 
+consists of a smart contract that stores a `idenitityCommitment` in a membership set.
+In order for a user to join the membership set,
+the user is required to make a transaction on the blockchain.
+A set of public keys are used to compute a stealth address,
+as desribed in [ERC-5564](https://eips.ethereum.org/EIPS/eip-5564).
+This specification is an implementation of the [ERC-5564](https://eips.ethereum.org/EIPS/eip-5564) scheme, 
+tailored to the curve that is used in the [32/RLN-V1](/spec/32) protocol.
+
+This can be used in couple of ways in applications:
+1. Applications can add users to the [32/RLN-V1](/spec/32) membership set in a batch.
+2. Users of the application can register other users to the [32/RLN-V1](/spec/32) membership set.
+
+This is useful when the prospective user does not have access to funds on the network that [32/RLN-V1](/spec/32) is deployed on.
+
+## Wire Format Specification 
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, 
+“NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+The wire formats are encoded using protobuf.
+The wire formats that SHOULD be used by implementors include two message types, 
+`RequestToJoin` and `JoinResponse`.
+
+### Request to Join
+The `RequestToJoin` message MUST be used by users joining a membership set with assistance from the application or 
+other users.
+The message should include `spending_public_key` and 
+a `viewing_public_key` that is generated by the member’s public key.
+
+The protobuf message description is:
+
+```protobuf
+
+syntax = "proto3";
+
+message RequestToJoin {
+  // The Lamport timestamp of the request  
+  uint64 clock = 1;
+
+  // The ENS name of the requester
+  string ens_name = 2;
+
+  // The chat id requested to join
+  string pubsub_topic = 3;
+
+  // The public key of the application
+  bytes app_key = 4;
+
+  // The display name of the requester
+  string display_name = 5;
+
+  // The spending public key of the requester
+  bytes spending_public_key = 6;
+
+  // The viewing public key of the requester
+  bytes viewing_public_key = 7;
+}
+
+```
+
+The `spending_public_key` and
+`viewing_public_key` MUST be included in the message.
+Both attributes create a stealth meta-address,
+as described in [ERC-5564](https://eips.ethereum.org/EIPS/eip-5564).
+- both SHOULD be a 32-byte hash.
+- both SHOULD be derived from the community member's public key.
+
+### Generate Stealth Address
+The application or user SHOULD generate a stealth address after the message is received.
+This address MUST be used to publish the requester's [32/RLN-V1](/spec/32) `identitycommitment` for the corresponding `app_key` memebership set.
```

Comment:
what is an `app_key`? Would be good to define that as pre-requisite.

Reply:
Changed to application membership set.